modules/ggl-file/src/file.c

```diff
@@ -578,3 +578,18 @@ GglError ggl_file_read_path(GglBuffer path, GglBuffer *content) {
 
     return GGL_ERR_OK;
 }
+
+GglError truncate_file(void *response_data) {
+    int fd = *(int *) response_data;
+
+    int ret;
+    do {
+        ret = ftruncate(fd, 0);
+    } while ((ret == -1) && (errno == EINTR));
```

Comment:
This errno might be overwritten by something else if used incorrectly. Not thread safe